tests/beta/queuing/test_beta_measure.py

```diff
@@ -250,3 +252,253 @@ def test_expand_no_observable(self):
+        with pytest.raises(ValueError, match="The jacobian method does not support"):
+            d_func(0.1)
+
+    def test_no_state(self):
+        """Test if an error is raised for devices that are not capable of returning the state,
+        such as default.gaussian"""
+        dev = qml.device("default.gaussian", wires=1)
+
+        @qnode(dev)
+        def func():
+            return state()
+
+        with pytest.raises(ValueError, match="The current device is not capable"):
+            func()
+
+    def test_no_state_but_capable(self, monkeypatch):
+        """Test if an error is raised for a device that incorrectly says it is capable of returning
+        the state. This is tested by changing the capability of default.gaussian."""
+        dev = qml.device("default.gaussian", wires=1)
+        capabilities = dev.capabilities().copy()
+        capabilities["returns_state"] = True
+
+        @qnode(dev)
+        def func():
+            return state()
+
+        with monkeypatch.context() as m:
+            m.setattr(DefaultGaussian, "capabilities", lambda *args, **kwargs: capabilities)
+            with pytest.raises(QuantumFunctionError, match="Returning the state is not supported"):
+                func()
+
+    @pytest.mark.usefixtures("skip_if_no_tf_support")
+    @pytest.mark.parametrize(
+        "device", ["default.qubit", "default.qubit.tf", "default.qubit.autograd"]
+    )
+    def test_devices(self, device, skip_if_no_tf_support):
+        """Test that the returned state is equal to the expected returned state for all of
+        PennyLane's built in statevector devices"""
+
+        dev = qml.device(device, wires=4)
+
+        @qnode(dev)
+        def func():
+            for i in range(4):
+                qml.Hadamard(i)
+            return state()
+
+        state_ev = func()
+        state_expected = 0.25 * np.ones(16)
+
+        assert np.allclose(state_ev, state_expected)
+        assert np.allclose(state_ev, dev.state)
+
+    @pytest.mark.usefixtures("skip_if_no_tf_support")
+    def test_gradient_with_passthru_tf(self, skip_if_no_tf_support):
+        """Test that the gradient of the state is accessible when using default.qubit.tf with the
+        backprop diff_method."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit.tf", wires=1)
+
+        @qnode(dev, interface="tf", diff_method="backprop")
+        def func(x):
+            qml.RY(x, wires=0)
+            return state()
+
+        x = tf.Variable(0.1, dtype=tf.complex128)
+
+        with tf.GradientTape() as tape:
+            result = func(x)
+
+        grad = tape.jacobian(result, x)
+        expected = tf.stack([-0.5 * tf.sin(x/2), 0.5 * tf.cos(x/2)])
+        assert np.allclose(grad, expected)
+
+    def test_gradient_with_passthru_autograd(self):
+        """Test that the gradient of the state is accessible when using default.qubit.autograd
+        with the backprop diff_method."""
+        from pennylane import numpy as npa
```

Comment:
very minor note: in all the other tests, we use the convention `from pennylane import numpy as anp` whenever we needed both vanilla numpy and autograd numpy. Probably best to keep to it here, to help find it in future refactors!

Reply:
Fixed!